packages/core/src/NodeExecuteFunctions.ts

```diff
@@ -3776,7 +3777,9 @@ export function getExecuteFunctions(
 				);
 				return dataProxy.getDataProxy();
 			},
-			async putExecutionToWait(waitTill: Date): Promise<void> {
+			async putExecutionToWait(waitTill: unknown): Promise<void> {
+				assertValidDate(waitTill);
```

Comment:
instead of putting this here (where we'll then validate all dates), we should put this check in the wait node only when using a user input date.